### PR TITLE
Update jet criteria

### PR DESCRIPTION
**Pull request by aperloff**
Update JetID
Update makeGoodJets to be cone size aware (i.e. minimum pT for AK8 jets is now 170).

### CODE REVIEW

File: TreeMaker/python/makeJetVars.py [hunk 1]
```diff
@@ -30,27 +30,55 @@ def makeMHTVars(self, process, JetTag, HTJetsTag, storeProperties, suff, MHTsuff
...
-        minChargedFraction        = cms.double(0),
-        maxChargedEMFraction      = cms.double(0.99),
-        jetPtFilter               = cms.double(30),
-        ExcludeLepIsoTrackPhotons = cms.bool(True),
-        JetConeSize               = cms.double(0.4),
-        SkipTag                   = SkipTag,
-        SaveAllJetsId             = True,
-        SaveAllJetsPt             = False, # exclude low pt jets from good collection
-    )
+    if self.localera == 'TM2016':
```
Comment (kpedro88) [inline TreeMaker/python/makeJetVars.py]: this is not how eras should be used
```
GoodJets = GoodJetsProducer.clone()
from TreeMaker.TreeMaker.TMEras import TMEras
TMEras.TM2016.toModify(GoodJets,[insert 2016 params here])
TMEras.TM2017.toModify(GoodJets,[insert 2017 params here])
```

**Comment by kpedro88** [comment]
also in the future please update branches by rebasing instead of merging

File: TreeMaker/python/makeJetVars.py (hunk 1 above)
Comment (kpedro88) [reply TreeMaker/python/makeJetVars.py]: ref: https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideCmsDriverEras

**Comment by aperloff** [comment]
Alright, the use of TMEras has been changed per your specifications. Also, I shall rebase in the future.

**Comment by aperloff** [comment]
Do you mean on line 37 and 56? Put a line break after "GoodJets,"?

**Comment by kpedro88** [comment]
yes

**Comment by aperloff** [comment]
Which indentation style do you like more?
```
    TMeras.TM2016.toModify(GoodJets,
                           TagMode                   = cms.bool(True),
                           JetTag                    = JetTag,
                           maxJetEta                 = cms.double(5.0),
                           minNconstituents          = cms.int32(1),
                           minNneutralsHF            = cms.int32(10),
                           minNcharged               = cms.int32(0),
                           maxNeutralFraction        = cms.double(0.99),
                           maxPhotonFraction         = cms.double(0.99),
                           maxPhotonFractionHF       = cms.double(0.90),
                           minChargedFraction        = cms.double(0),
                           maxChargedEMFraction      = cms.double(0.99),
                           jetPtFilter               = cms.double(30),
                           ExcludeLepIsoTrackPhotons = cms.bool(True),
                           JetConeSize               = cms.double(jetConeSize),
                           SkipTag                   = SkipTag,
                           SaveAllJetsId             = True,
                           SaveAllJetsPt             = False, # exclude low pt jets from good collection
                           )
    TMeras.TM2017.toModify(GoodJets,
                                    TagMode                   = cms.bool(True),
                                    JetTag                    = JetTag,
                                    maxJetEta                 = cms.double(5.0),
                                    minNconstituents          = cms.int32(1),
                                    minNneutralsHE            = cms.int32(2),
                                    minNneutralsHF            = cms.int32(10),
                                    minNcharged               = cms.int32(0),
                                    maxNeutralFraction        = cms.double(0.90),
                                    maxNeutralFractionHE      = cms.double(1.00), #Turned off as not needed for the tight WP
                                    minNeutralFractionHF      = cms.double(0.02),
                                    maxPhotonFraction         = cms.double(0.90),
                                    minPhotonFractionHE       = cms.double(0.02),
                                    maxPhotonFractionHE       = cms.double(0.99),
                                    maxPhotonFractionHF       = cms.double(0.90),
                                    minChargedFraction        = cms.double(0.00),
                                    maxChargedEMFraction      = cms.double(1.00), #Turned off as not needed for the tight WP
                                    jetPtFilter               = cms.double(170 if jetConeSize==0.8 else 30),
                                    ExcludeLepIsoTrackPhotons = cms.bool(True),
                                    JetConeSize               = cms.double(jetConeSize),
                                    SkipTag                   = SkipTag,
                                    SaveAllJetsId             = True,
                                    SaveAllJetsPt             = False, # exclude low pt jets from good collection
                          )
```

**Comment by kpedro88** [comment]
neither
```
    TMeras.TM2016.toModify(GoodJets,
        TagMode                   = cms.bool(True),
    )
```

**Comment by aperloff** [comment]
Style changed.

**Comment by kpedro88** [comment]
@aperloff I realized we need to fix this setting as well:
https://github.com/TreeMaker/TreeMaker/blob/d857e38486b3a0daa82c2acf655c4c059ed250ce/TreeMaker/python/makeJetVars.py#L80

**Comment by aperloff** [comment]
So it should be 20 for 2016, but should it stay the same for 2017? Should it also filter AK8 jets at 170GeV? Once I know this I'll put in a new PR.

**Comment by kpedro88** [comment]
Let's keep it at 20 for 2017 as well, for now. This was needed because of some silly fastsim filter, but I have no idea if it will still be needed in 2017 or not. (There aren't any 2017 fastsim samples yet...) Yes, we should also put the 170 cut on AK8 jets for fastsim.

**Pull request merged**